### PR TITLE
[processing] Set z and m values algorithms

### LINKED ISSUE #2825: [FEATURE][processing] native drape features to z/m algorithms
Original commit: https://github.com/qgis/QGIS/commit/d5ce6dcb1df485feb1fc1e9ed7f6eb1dad5d2f49 by nyalldawson

Sets vertex z/m values to values sampled from a raster band.

Values can optionally be scaled using a (data definable) scale value.

### PR DESCRIPTION
**Pull request by ghtmtt**
fixes #2825

### CODE REVIEW

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 1]
```diff
@@ -815,6 +815,48 @@ Outputs
...
+Parameters
+..........
+``Input layer`` [vector: any]
+  Input vector layer to set the z-values to
+
+``Raster layer`` [raster]
+  Raster layer to take the z-values from
+
+``Band number`` [raster band]
+  If the raster is multiband choose the band you want to take the z-values from
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: `The raster band to take the z-values from if the raster is multiband`? I know both descriptions look alike but the idea here is that description describes what the feature is and not the user behavior. Also,  it follows the other parameters description style (eg, `Raster layer to take the z-values from`).

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 2]
```diff
@@ -815,6 +815,48 @@ Outputs
...
+
+``Scale`` [number | dataDefined]
+  Scaling value
+
+  Default: *1.0*
+
+Output
+......
+
+``Updated`` [vector: any]
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Wonder why the output is called "updated"? not sure the z field has to exist beforehand.

File: source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst [hunk 3]
```diff
@@ -815,6 +815,48 @@ Outputs
...
+  Vector layer in output with the updated z-values extracted
+
+See also
+........
+:ref:`qgissetmfromraster`
```
Comment (DelazJ) [inline source/docs/user_manual/processing_algs/qgis/vectorgeometry.rst]: Do you think we could add the `set z values` here

**Comment by ghtmtt** [comment]
> Wonder why the output is called "updated"? not sure the z field has to exist beforehand.

no idea.. I'll let the name as the main algorithm and change it when needed

**Comment by ghtmtt** [comment]
+1 to merge. Thanks @DelazJ

**Pull request merged**